platform/darwin/src/MGLOfflineStorage.mm

```diff
@@ -357,6 +359,13 @@ - (void)addPackForRegion:(id <MGLOfflineRegion>)region withContext:(NSData *)con
         [[strongSelf mutableArrayValueForKey:@"packs"] addObject:pack];
         if (completion) {
             completion(pack, error);
+            
+            [MGLMapboxEvents pushEvent:MMEEventTypeOfflineDownload withAttributes:@{
+                                                                                    @"shapeForOfflineRegion": NSStringFromClass(pack.region),
+                                                                                    @"minZoom": @0, // get zoom level
+                                                                                    @"maxZoom": @0, // get zoom level
+                                                                                    @"sources": @[] // ?
+                                                                                    }];
```

Comment:
It’s true that MGLOfflineStorage doesn’t have access to the runtime styling models, but the first resource that mbgl::OfflineManager downloads for any pack is the style JSON file. That file tells mbgl about all the sources, so at that point it could execute a callback that tells the SDK the URLs of the sources involved.

Note that some sources aren’t backed by TileJSON files, instead being inlined in the style. Inlined sources aren’t as interesting for analytical purposes, since they tend to be one-offs. Either way, there’s an mbgl::TileSet object that the SDK could be told about, and the SDK could then decide what to send as part of the event.